lib/fog/vsphere/requests/compute/list_child_snapshots.rb

```diff
@@ -0,0 +1,68 @@
+module Fog
+  module Compute
+    class Vsphere
+      class Real
+        def list_child_snapshots(snapshot, opts = {})
+          s = Hash === snapshot ?
+            Snapshot.new(snapshot.update(service: self)) : snapshot
+          s.tree_node.childSnapshotList.map do |snap|
+            item = child_snapshot_info(snap, s)
+            [
+              item,
+              opts[:recursive] ? list_child_snapshots(item, opts) : nil
+            ]
+          end.flatten.compact
+        end
+
+        protected
+
+          def child_snapshot_info(snap_tree, parent_snap)
+            {
```

Comment:
It is implemented the same way as other requests in vsphere (see [list_networks.rb](https://github.com/fog/fog/blob/master/lib/fog/vsphere/requests/compute/list_networks.rb) or [list_datastores.rb](https://github.com/fog/fog/blob/master/lib/fog/vsphere/requests/compute/list_datastores.rb).

Furthermore it would not be easy to implement it in this way, the properties are composed from multiple inputs: snap_tree, parent_snap.